File: Data Structures & Algorithms/max-consecutive-ones-iii/submission-3.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        
        int n = nums.length;
        int l = 0, r = 0;
        int zc = 0;
        int ans = 0;

        while( r < n)
        {
            if(nums[r] == 0)
                zc++;
            
            while( l <= r && zc > k)
            {
                if(nums[l] == 0)
                    zc--;
                
                l++;
            }

            int len = r - l + 1;
            ans = Math.max(ans, len);
            r++;
        }

        return ans;
    }
}

File: Data Structures & Algorithms/max-consecutive-ones-ii/submission-3.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        zeroCount,n = 0,len(nums)
        ans = 0
        l,r = 0,0
        for i in range(n):
            # Incr zero count
            if nums[i] == 0:
                zeroCount += 1
            # zero count is > 1, slide the window
            while l <= r and zeroCount > 1:
                if nums[l] == 0:
                    zeroCount -= 1
                l += 1
            # Take teh max window as ans
            ans = max(ans,r-l+1)
            r += 1
        
        return ans